lib/diagnostics_channel.js

```diff
@@ -105,27 +188,17 @@ function channel(name) {
   }
```

Comment:
Placed where it is, it can skip the extra checking cost in the case where it already exists--the object (now becoming a map) should only be able to contain a match if the channel _creation_ path adds an entry _after_ the validation. Though admittedly the performance in this case _probably_ doesn't really matter. Do you feel it should be changed anyway? I'm fine either way.

Reply:
In the actual case it's don't care so feel free to keep as is.
In general I find it fishy if a type check (or any other validation) is done after use.